commands/promote.js

```diff
@@ -61,6 +63,45 @@ function * run (context, heroku) {
       }
     })
   }))
+
+  let releasePhase = (yield heroku.get(`/apps/${app}/formation`))
+    .find((formation) => formation.type === 'release')
+
+  if (releasePhase) {
+    yield cli.action('Checking release phase', co(function * () {
+      let releases = (yield heroku.get(`/apps/${app}/releases`))
+          .sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at))
+      let attach = releases.find((release) => release.description.includes('Attach DATABASE'))
+      let detach = releases.find((release) => release.description.includes('Detach DATABASE'))
+
+      if (!attach || !detach) {
+        throw new Error(`Unable to find Detach and Attach DATABASE releases ${cli.color.app(app)}`)
+      }
+
+      let [attachId, detachId] = [attach.id, detach.id]
+      while (true) {
+        let attach = yield heroku.get(`/apps/${app}/releases/${attachId}`)
+        if (attach && attach.status === 'succeeded') {
+          let msg = 'pg:promote succeeded.'
+          let detach = yield heroku.get(`/apps/${app}/releases/${detachId}`)
+          if (detach && detach.status === 'failed') {
+            msg += ` It is safe to ignore the failed ${detach.description} release.`
+          }
+          return cli.action.done(msg)
+        } else if (attach && attach.status === 'failed') {
+          let msg = `pg:promote failed because ${attach.description} release was unsuccessful. Your application is currently running `
```

Comment:
So immediately after seeing this, a customer will think, "Ok, how do I fix it?" I know a full explanation is beyond the scope of this PR, but should we give guidance like "Check your release phase logs" or something?

Reply:
yeah I think we don't wanna that far (explaining everything in this PR), but having "check your release phase logs" sounds like a good thing to add.

Reply:
added `' Check your release phase logs for failure causes.'` to the message